### ISSUE TITLE
unnecessary_getters_setters warns for getter even though setter is necessary

### ISSUE THREAD

**Issue by tjarvstrand** [dart-lang/linter#3759]
Eg. when implementing the controller-pattern in Flutter, it is common to expose one or more members with setters that notify listeners and getters that don't have any additional logic. With `unnecessary_getters_setters` this results in a warning.

**To Reproduce**
```
class MyClass {
  int _value = 0;
  MyClass();

  int get value => _value; // unnecessary_getters_setters warns here
  set value(int newValue) {
    // ...do important stuff here
    _value = newValue;
  }
}
```

**Expected behavior**
unnecessary_getters_setters should not warn for this.
